src/server/scripts/Outland/BlackTemple/boss_illidan.cpp

```diff
@@ -50,2180 +37,2571 @@ enum Say
+                    events.ScheduleEvent(EVENT_FLAME_BURST, Seconds(6), group);
+                    events.ScheduleEvent(EVENT_SHADOW_DEMON, Seconds(18), Seconds(30), group);
+                case GROUP_PHASE_4:
+                    ScheduleEvents(GROUP_PHASE_3, group);
+                    events.ScheduleEvent(EVENT_FRENZY, Seconds(40), group);
+                    break;
+                default:
+                    break;
             }
         }
 
-        void EnrageCheck()
+        void JustSummoned(Creature* summon) override
         {
-            if (Creature* glaive = ObjectAccessor::GetCreature(*me, GlaiveGUID))
+            BossAI::JustSummoned(summon);
+            switch (summon->GetEntry())
             {
-                if (!me->IsWithinDistInMap(glaive, FLAME_ENRAGE_DISTANCE))
-                {
-                    glaive->InterruptNonMeleeSpells(true);
-                    DoCast(me, SPELL_FLAME_ENRAGE, true);
-                    DoResetThreat();
-                    if (SelectTarget(SELECT_TARGET_RANDOM, 0))
+                case NPC_ILLIDARI_ELITE:
+                    if (Creature* akama = instance->GetCreature(DATA_AKAMA))
                     {
-                        me->AddThreat(me->GetVictim(), 5000000.0f);
-                        AttackStart(me->GetVictim());
+                        summon->CombatStart(akama);
+                        summon->AddThreat(akama, 1000.0f);
                     }
-                }
-                else if (!me->HasAura(SPELL_AZZINOTH_CHANNEL))
-                {
-                    glaive->CastSpell(me, SPELL_AZZINOTH_CHANNEL, false);
-                    me->RemoveAurasDueToSpell(SPELL_FLAME_ENRAGE);
-                }
+                    break;
+                case NPC_CAGE_TRIGGER_1:
+                    summon->CastSpell(summon, SPELL_CAGED_TRIGGER_1, true);
+                    break;
+                case NPC_CAGE_TRIGGER_2:
+                    summon->CastSpell(summon, SPELL_CAGED_TRIGGER_2, true);
+                    break;
+                case NPC_CAGE_TRIGGER_3:
+                    summon->CastSpell(summon, SPELL_CAGED_TRIGGER_3, true);
+                    break;
+                case NPC_CAGE_TRIGGER_4:
+                    summon->CastSpell(summon, SPELL_CAGED_TRIGGER_4, true);
+                    break;
+                case NPC_CAGE_TRIGGER_5:
+                    summon->CastSpell(summon, SPELL_CAGED_TRIGGER_5, true);
+                    break;
+                case NPC_CAGE_TRIGGER_6:
+                    summon->CastSpell(summon, SPELL_CAGED_TRIGGER_6, true);
+                    break;
+                case NPC_CAGE_TRIGGER_7:
+                    summon->CastSpell(summon, SPELL_CAGED_TRIGGER_7, true);
+                    break;
+                case NPC_CAGE_TRIGGER_8:
+                    summon->CastSpell(summon, SPELL_CAGED_TRIGGER_8, true);
+                    break;
```

Comment:
@offl i did: https://gist.github.com/Keader/d1cbbc3ecd161b0a4eb9f12cd1750fd4
but not work :(
Npcs only start cast if i put .gm on